### ISSUE TITLE
mySQL forced restart logic

### ISSUE THREAD

**Issue by angel0d** [omega8cc/boa#799]
hi,

I am trying to understand when and how mySQL is killed/restarted

I studied mysql_proc_kill and mysql_proc_control in minute.sh but I think I need translation

the mysql.forced.restart.log reports

mysql CPU:328 MAXSQLCPU:195 Rl START:18:52 TIME:75:06 151014-191655
mysql CPU:282 MAXSQLCPU:195 Rl START:19:16 TIME:65:23 151014-194022
mysql CPU:279 MAXSQLCPU:195 Rl START:19:39 TIME:22:30 151014-194848
mysql CPU:442 MAXSQLCPU:195 Rl START:19:48 TIME:181:06 151014-203005
mysql CPU:407 MAXSQLCPU:195 Rl START:20:29 TIME:63:43 151014-204606
mysql CPU:369 MAXSQLCPU:195 Rl START:20:45 TIME:69:25 151014-210517
mysql CPU:205 MAXSQLCPU:195 Rl START:08:41 TIME:167:16 151015-100422
mysql CPU:206 MAXSQLCPU:195 Rl START:10:03 TIME:234:32 151015-115836
mysql CPU:224 MAXSQLCPU:195 Rl START:11:58 TIME:283:57 151015-140520
mysql CPU:196 MAXSQLCPU:195 Rl START:14:04 TIME:251:00 151015-161408

we indeed had high load during the last days but we 'kind of' fixed it and I am trying to find out if there is a way that temporary peaks can be tolerated (see last log entry)

thanks in advance

Angelo

**Comment by omega8cc** [comment]
This is simply way too high CPU usage for mysql. You need to either disable this procedure in minute.sh or fix the sites hosted, so they will use db server without trying to kill the system. Maybe you have disabled Redis or it is not used? If not, your system is seriously overloaded.

**Comment by angel0d** [comment]
one more detail I missed to mention is that it is a particular VM by which I mean that it is a Linode of 48GB with 16 'real' (pass through) CPUs

and while I see the 3 following entries in .barracuda.cnf

_CPU_CRIT_RATIO=9 #------------- Max load per CPU core before killing PHP/Drush
_CPU_MAX_RATIO=6 #-------------- Max load per CPU core before disabling Nginx
_CPU_SPIDER_RATIO=3 #----------- Max load per CPU core before blocking spiders

and have found and read some legacy issues on the matter and as far as I cannot find anything related to SQL sever ...

... I am still wondering if instead of circumventing  the issue we could control it and also if there might be a mis-calculation of the available resource

I have confirmed that with 'cat /proc/cpuinfo' all 16 cores show ok

**Comment by omega8cc** [comment]
These variables are not used in this case. Instead we hardcode it for SQL and FPM monitoring, since otherwise, especially with multicore systems, you can freeze the system no matter how many cores you have if the mysql CPU usage goes over 200:

```
sub cpu_count_load
{
  local($PROCS) = `grep -c processor /proc/cpuinfo`;
  chomp($PROCS);
  $MAXSQLCPU = $PROCS."00";
  $MAXFPMCPU = $PROCS."00";
  if ($PROCS > 2)
  {
    $MAXSQLCPU = 200;
  }
  $MAXSQLCPU = $MAXSQLCPU - 5;
  $MAXFPMCPU = $MAXFPMCPU - 5;
}
```

Note that this logic is used in the `/var/xdrago/proc_num_ctrl.cgi`

What is the result of `grep -c processor /proc/cpuinfo` on your system?

**Comment by angel0d** [comment]
as I wrote before "I have confirmed that with 'cat /proc/cpuinfo' all 16 cores show ok" and now confirming by grep : 

root@live:~# grep -c processor /proc/cpuinfo
16

it seems to me that the upper limit should be 1600 and not 200

**Comment by omega8cc** [comment]
No, the upper limit must be 200. We have confirmed this many times and the limit has been carefully tuned to save system from crashing. We have not invented this feature/limit. We have introduced it and tuned to exactly this value as a result of recovering some remotely supported systems from crash caused by mysql process going out of control. When you allow more than that, it would be useless protection.

That said, we could add a control file to turn this protection off.
